src/axom/sidre/core/Group.hpp

```diff
@@ -154,6 +154,38 @@ class Group
    */
   char getPathDelimiter() const { return s_path_delimiter; }
 
+  /*!
+   * \brief static method to get valid protocols for Group I/O methods.
+   *
+   * Only protocols that work for both input and output are returned.
+   */
+  static const std::vector<std::string>& getValidIOProtocols()
+  {
+    // The first time this is called, fill the vector with valid protocols.
+    if(s_io_protocols.empty())
+    {
+#ifdef AXOM_USE_HDF5
+      s_io_protocols.push_back("sidre_hdf5");
+      s_io_protocols.push_back("conduit_hdf5");
+#endif
+      s_io_protocols.push_back("sidre_json");
+      s_io_protocols.push_back("sidre_conduit_json");
+      s_io_protocols.push_back("conduit_bin");
+      s_io_protocols.push_back("conduit_json");
+      s_io_protocols.push_back("json");
+    }
+
+    return s_io_protocols;
```

Comment:
Suggestion: Would it be better if this function created the vector and then returned it?
(Although it's unlikely), the way it's currently implemented might allow the vector to be modified elsewhere in `sidre::Group`

Reply:
I have a similar idea to protect it from change, and I will add it before the merge.

Reply:
I changed the declaration of `s_io_protocols` to `static const` and gave it brace initialization so that it doesn't have be filled on the first `get` call.